sys/zuse.hoon

```diff
@@ -2966,6 +2966,298 @@
       ::  alice decrypts with same key
       `@t`(dy.ali secret-key crypted-msg)
     --  ::test
+  ::                                                    ::
+  ::::                    ++keccak:crypto               ::  (2b7) keccak family
+    ::                                                  ::::
+  ++  keccak
+    |%
+    ::
+    ::  keccak
+    ::
+    ++  keccak-224  |=(a=@ (keccak 1.152 448 224 a))
+    ++  keccak-256  |=(a=@ (keccak 1.088 512 256 a))
+    ++  keccak-384  |=(a=@ (keccak 832 768 384 a))
+    ++  keccak-512  |=(a=@ (keccak 576 1.024 512 a))
+    ::
+    ++  keccak  (cury (cury hash keccak-f) padding-keccak)
+    ::
+    ++  padding-keccak  (multirate-padding 0x1)
+    ::
+    ::  sha3
+    ::
+    ++  sha3-224  |=(a=@ (sha3 1.152 448 224 a))
+    ++  sha3-256  |=(a=@ (sha3 1.088 512 256 a))
+    ++  sha3-384  |=(a=@ (sha3 832 768 384 a))
+    ++  sha3-512  |=(a=@ (sha3 576 1.024 512 a))
+    ::
+    ++  sha3  (cury (cury hash keccak-f) padding-sha3)
+    ::
+    ++  padding-sha3  (multirate-padding 0x6)
+    ::
+    ::  shake
+    ::
+    ++  shake-128  |=([o=@ud i=@] (shake 1.344 256 o i))
+    ++  shake-224  |=([o=@ud i=@] (shake 1.088 512 o i))
+    ::
+    ++  shake  (cury (cury hash keccak-f) padding-shake)
+    ::
+    ++  padding-shake  (multirate-padding 0x1f)
+    ::
+    ::  rawshake
+    ::
+    ++  rawshake-128  |=([o=@ud i=@] (rawshake 1.344 256 o i))
+    ++  rawshake-224  |=([o=@ud i=@] (rawshake 1.088 512 o i))
+    ::
+    ++  rawshake  (cury (cury hash keccak-f) padding-rawshake)
+    ::
+    ++  padding-rawshake  (multirate-padding 0x7)
+    ::
+    ::  core
+    ::
+    ++  hash
+      |=  $:  per=$-(@ud $-(@ @))
```

Comment:
These args deserve individual comments.

Reply:
Also, does the input need to be supplied with a specific endianness, or other requirements?